### ISSUE TITLE
横条纹问题

### ISSUE THREAD

**Issue by Doncic7** [OpenImagingLab/FlashVSR#17]
<img width="706" height="1225" alt="Image" src="https://github.com/user-attachments/assets/278ad53d-8ac8-4ee5-9923-fc6ac0524cfb" />

其他三个例子都是正常的，只有example2出现横条纹问题，不知道是不是因为图片比例问题？
我的设置均为默认代码设置
运行脚本为tiny.py  在1张A100上运行
block_sparse_attn        0.0.1
torch                    2.6.0+cu124
torchvision              0.21.0+cu124
transformers             4.46.2
diffsynth                1.1.7

seed, scale, dtype, device = 0, 4.0, torch.bfloat16, 'cuda'
    sparse_ratio = 2.0 

video = pipe(
            prompt="", negative_prompt="", cfg_scale=1.0, num_inference_steps=1, seed=seed,
            LQ_video=LQ, num_frames=F, height=th, width=tw, is_full_block=False, if_buffer=True,
            topk_ratio=sparse_ratio*768*1280/(th*tw), 
            kv_ratio=3.0,
            local_range=11,  # Recommended: 9 or 11. local_range=9 → sharper details; 11 → more stable results.
            color_fix = False,
        )
color_fix   为false和true的情况下都进行了实验，还需要什么信息我都可以提供，麻烦博主关注下这个情况

**Comment by zhuang2002** [comment]
You can contact me by email first, then we can connect on WeChat — debugging online is too slow.

**Comment by Doncic7** [comment]
> You can contact me by email first, then we can connect on WeChat — debugging online is too slow.

I have sent a email,please check it when you're free,Thankyou!

**Comment by kelisiyaer-ctrl** [comment]
mark

**Comment by kelisiyaer-ctrl** [comment]
> You can contact me by email first, then we can connect on WeChat — debugging online is too slow.

I seem to have discovered a new pattern: this phenomenon occurs very easily when I continuously infer a video folder. However, when I infer the problematic video alone, it magically disappears. I will further explore this issue. I suspect it is related to seed reset or caching.

**Comment by kelisiyaer-ctrl** [comment]
> > You can contact me by email first, then we can connect on WeChat — debugging online is too slow.
> 
> I seem to have discovered a new pattern: this phenomenon occurs very easily when I continuously infer a video folder. However, when I infer the problematic video alone, it magically disappears. I will further explore this issue. I suspect it is related to seed reset or caching.

I can now stably reproduce this issue: when I first perform inference on a landscape video, and then the next file is a portrait video, the problem occurs. The same applies in reverse—videos with a changed aspect ratio always develop block artifacts.

**Comment by kelisiyaer-ctrl** [comment]
> You can contact me by email first, then we can connect on WeChat — debugging online is too slow.

When inferring on a folder, simply running pipe = init_pipeline() and torch.cuda.empty_cache() each time can resolve this block artifact issue. This indicates that there are certain acceleration mechanisms or caches within the pipeline that interfere with normal inference, which requires further investigation by the authors. I will also conduct relevant explorations subsequently.

**Comment by zhuang2002** [comment]
<img width="1573" height="392" alt="Image" src="https://github.com/user-attachments/assets/dfa3ca4f-2a4f-43a5-9f6f-037d4326f27e" />

目前问题已经解决。原因是 `local_attention_mask` 的更新逻辑不完整，以前只在 **分辨率面积变化** 时才更新，但没有处理 **竖屏↔横屏切换或 H×W 面积不变但形状变化** 的情况，导致掩码被错误复用。该问题已修复。

如果问题已解决，你可以关闭这个 issue。

---

The issue has been resolved. The root cause was an incomplete update logic for `local_attention_mask`: it was only refreshed when the **resolution area changed**, but did not handle **aspect ratio changes (portrait ↔ landscape) or shape changes with the same H×W area**, which caused the mask to be incorrectly reused. This has now been fixed.

If everything looks good on your side, feel free to close this issue.

@Doncic7 @kelisiyaer-ctrl @laodar @Franc1sCai @SekiroRong
